web/impact/impact/tests/test_graphql.py

```diff
@@ -582,10 +548,98 @@ def test_query_for_user_without_confirmed_mentor_program_families(self):
+                    programRoles
+                }}
+            }}
+            """.format(id=user.id)
+
+            self._assert_error_in_response(query, NOT_ALLOWED_ACCESS_MESSAGE)
+
+    def test_user_without_role_can_view_staff_entrepreneur_profile(self):
+        current_user = self._expert_user()
+        user = EntrepreneurFactory()
+        UserRoleContext(UserRole.STAFF, user=user)
+        query = """
+            query{{
+                entrepreneurProfile(id:{id}) {{
+                    user{{lastName}}
+                }}
+            }}
+        """.format(id=user.id)
+        expected_json = {
+            'data': {
+                'entrepreneurProfile': {
+                    'user': {
+                        'lastName': user.last_name
+                    }
+                }
+            }
+        }
+        self._assert_response_equals_json(
+            query, expected_json, email=current_user.email)
+
+    def test_allowed_user_can_view_mentor_entrepreneur_profile(self):
+        allowed_user_roles = [
+            UserRole.FINALIST,
+            UserRole.AIR,
+            UserRole.MENTOR,
+            UserRole.PARTNER,
+            UserRole.ALUM
+        ]
+        for role in allowed_user_roles:
+            current_user = self._expert_user(role)
+            user = EntrepreneurFactory()
+            UserRoleContext(UserRole.MENTOR, user=user)
+            query = """
+                query{{
+                    entrepreneurProfile(id:{id}) {{
+                        user{{lastName}}
+                    }}
+                }}
+            """.format(id=user.id)
+            expected_json = {
+                'data': {
+                    'entrepreneurProfile': {
+                        'user': {
+                            'lastName': user.last_name
+                        }
+                    }
+                }
+            }
+            self._assert_response_equals_json(
+                query, expected_json, email=current_user.email)
+
+    def _assert_response_equals_json(self, query, expected_json, email=None):
+        with self.login(email=email or self.basic_user().email):
             response = self.client.post(self.url, data={'query': query})
             self.assertJSONEqual(
                 str(response.content, encoding='utf8'),
                 expected_json
             )
+
+    def _assert_error_in_response(self, query, error_message):
+        with capture_stderr(self.client.post,
+                            self.url,
+                            data={'query': query}) as (response, _):
+            error_messages = [x['message']
+                              for x in response.json()['errors']]
+        self.assertIn(error_message, error_messages)
+
+    def _expert_user(self, role=None):
+        user = ExpertFactory()
+        if role:
+            user_role = get_user_role_by_name(role)
+            program_role = ProgramRoleFactory.create(user_role=user_role)
```

Comment:
we do not need to use - and prefer not to use - the `.create` syntax, since `ProgramRoleFactory(user_role=user_role)` means and does exactly the same thing.